Add files via upload

AlexeyVertoprakhov opened the pull request.

AlexeyVertoprakhov committed:
Add files via upload

AlexeyVertoprakhov committed:
Add files via upload

AlexeyVertoprakhov committed:
Update task_1.py

AlexeyVertoprakhov committed:
Update task_2.py

AlexeyVertoprakhov committed:
Update task_3.py

AlexeyVertoprakhov committed:
Update task_4.py

AlexeyVertoprakhov committed:
Update task_5.py

AlexeyVertoprakhov committed:
Update task_6.py

AlexeyVertoprakhov committed:
Update task_7.py

AlexeyVertoprakhov committed:
Update task_4.py

AlexeyVertoprakhov committed:
Update task_4.py

AlexeyVertoprakhov committed:
Update task_1.py

AlexeyVertoprakhov committed:
Update task_1.py

AlexeyVertoprakhov committed:
Update task_2.py

AlexeyVertoprakhov committed:
Update task_1.py

AlexeyVertoprakhov committed:
Update task_1.py

AlexeyVertoprakhov committed:
Update task_3.py

AlexeyVertoprakhov committed:
Update task_4.py

AlexeyVertoprakhov committed:
Update task_7.py

AlexeyVertoprakhov committed:
Update task_1.py

AlexeyVertoprakhov committed:
Update task_2.py

AlexeyVertoprakhov committed:
Update task_3.py

AlexeyVertoprakhov committed:
Update task_4.py

DmitryChitalov commented on Урок 2. Практическое задание/task_1.py:
```diff
@@ -1,30 +1,61 @@
+print("Сумма двух чисел равна : ", sum(a, b))
+print("Разность двух чисел равна : ", residual(a, b))
+print("Произведение двух чисел будет : ", product(a, b))
+print("Частное двух чисел будет : ", remaind(a, b))
```
решение не увязано в единый скрипт
посмотрите пример

DmitryChitalov commented on Урок 2. Практическое задание/task_2.py:
```diff
@@ -1,18 +1,24 @@
+def count_digits(num):
+    return Counter('нечетных чисел' if int(d) % 2 else 'четных чисел' for d in str(num))
+
+print (count_digits(int(input("Введите любое целое число : "))))
```
лишний пробел и Counter мы пока не знаем
и рекурсии здесь нет

DmitryChitalov commented on Урок 3. Практическое задание/task_1.py:
```diff
@@ -1,12 +1,28 @@
+    else:
+        print(f'Для n={n} - False')
+        break
+    n += 1
```
Реализуйте заполнение списка и словаря, сделайте замеры и сделайте выводы,
не вижу этого у вас

DmitryChitalov commented on Урок 3. Практическое задание/task_2.py:
```diff
@@ -15,3 +15,48 @@
+key_from_storage = storage[32:]
+
+import hashlib
+
```
для чего импорт еще раз?
почему импорт посреди кода?
нарушение пеп

DmitryChitalov commented on Урок 3. Практическое задание/task_3.py:
```diff
@@ -14,3 +14,32 @@
+    for j in range(len(S) - 1 if i == 0 else len(S), i, -1):
+        subs_set.add(hash(S[i:j]))
+
+print("Количество различных подстрок в этой строке:", len(subs_set))
```
нет вычисления хешей

DmitryChitalov commented on Урок 3. Практическое задание/task_4.py:
```diff
@@ -7,4 +7,22 @@
+            return self.fun(*args, **kwargs)
+@cache
+def get_url_src(url):
+    return urllib.urlopen(url).read()
```
нет примера работы скрипта с передачей реальных ссылок

AlexeyVertoprakhov committed:
Delete task_1.py

"""
Задание 1.
Для каждой из трех задач выполнить следующее:
1) для каждой инструкции рядом в комментарии определите сложность этой инструкции
2) определите сложность алгоритма в целом
укажите сложность непосредственно в этом файле
точки, где нужно поработать вам, отмечены знаком '!!!'
Примечание:
Если у вас возникают сложности, постарайтесь подумать как можно решить задачу,
а не писать "мы это не проходили)".
Алгоритмизатор должен развивать мышление, а это прежде всего практика.
А без столкновения со сложностями его не развить.
"""

import random


#############################################################################################
def check_1(lst_obj):
    """Функция должна создать множество из списка.
    Алгоритм 3:
    Создать множество из списка
    Сложность: O(n) .
    """
    lst_to_set = set(lst_obj)  # n
    return lst_to_set  # 1


#############################################################################################
def check_2(lst_obj):
    """Функция должная вернуть True, если все элементы списка различаются.
    Алгоритм 1:
    Проходимся по списку и для каждого элемента проверяем,
    что такой элемент отстутствует
    в оставшихся справа элементах
    Сложность: O(n).
    """
    for j in range(len(lst_obj)):  # n
        if lst_obj[j] in lst_obj[j + 1:]:  # n
            return False  # 1
    return True  # 1


#############################################################################################
def check_3(lst_obj):
    """Функция должная вернуть True, если все элементы списка различаются.
    Алгоритм 2:
    Вначале выполним для списка сортировку, далее, сравниваем элементы попарно
    Если присутствуют дубли, они будут находиться рядом.
    Сложность: O(n)
    """
    lst_copy = list(lst_obj)  # n
    lst_copy.sort()  # n
    for i in range(len(lst_obj) - 1):  # n
        if lst_copy[i] == lst_copy[i + 1]:  # 1
            return False  # 1
    return True  # 1


#############################################################################################


for j in (50, 500, 1000, 5000, 1000):
    # Из 100000 чисел возьмем 'j' случайно выбранных
    # Всего 10 тыс. чисел

    lst = random.sample(range(-100000, 100000), j)

print(check_1(lst))
print(check_2(lst))
print(check_3(lst))

AlexeyVertoprakhov committed:
Update task_2.py

AlexeyVertoprakhov committed:
Update task_3.py

AlexeyVertoprakhov committed:
Update task_4.py

AlexeyVertoprakhov committed:
Update task_5.py

AlexeyVertoprakhov committed:
Update task_6.py

AlexeyVertoprakhov committed:
Update task_7.py

AlexeyVertoprakhov committed:
Update task_1.py

AlexeyVertoprakhov committed:
Update task_2.py

AlexeyVertoprakhov committed:
Update task_3.py

AlexeyVertoprakhov committed:
Update task_4.py
